packages/kit/scripts/extract-types.js

```diff
@@ -19,59 +20,72 @@ function get_types(code, statements) {
 	/** @type {Extracted[]} */
 	const types = [];
 
-	for (const statement of statements) {
-		if (
-			ts.isClassDeclaration(statement) ||
-			ts.isInterfaceDeclaration(statement) ||
-			ts.isTypeAliasDeclaration(statement) ||
-			ts.isModuleDeclaration(statement) ||
-			ts.isVariableStatement(statement) ||
-			ts.isFunctionDeclaration(statement)
-		) {
-			const name_node = ts.isVariableStatement(statement)
-				? statement.declarationList.declarations[0]
-				: statement;
-
-			// @ts-ignore no idea why it's complaining here
-			const name = name_node.name?.escapedText;
-
-			let start = statement.pos;
-			let comment = '';
-
-			// @ts-ignore i think typescript is bad at typescript
-			if (statement.jsDoc) {
-				// @ts-ignore
-				comment = statement.jsDoc[0].comment;
-				// @ts-ignore
-				start = statement.jsDoc[0].end;
+	if (statements) {
```

Comment:
personal preference, but I'd write this as shown below. right now, it's a bit hard to see where the `if` block ends and what comes after it since it doesn't fit on one screen. it'd also save having to have the code inside indented as much
```
if (!statements) {
  return { types, exports };
}
```

Reply:
my personal preference goes the other way 😆 i know you love an early return, i much prefer positive conditions over negative ones — way fewer mental contortions — and not having duplicated identical `return` statements